### PR TITLE
Feat: 2.0

### PR DESCRIPTION
**Pull request by bpolaszek**
Hello @simPod,

Do you think we're ready to go for 2.0?

Thank you,
Ben

### CODE REVIEW

**Comment by simPod** [comment]
Yes, the only thing that came to my mind was this https://github.com/bpolaszek/php-iterable-functions/pull/33#discussion_r582746034

**Comment by bpolaszek** [comment]
> Yes, the only thing that came to my mind was this [#33 (comment)](https://github.com/bpolaszek/php-iterable-functions/pull/33#discussion_r582746034)

I must admit I don't really understand the concern. 
IMO functions are one-liners, if you want to chain calls, better use the `iterable()` fluent interface.
Can you be more specific about what harm this could cause?

**Comment by simPod** [comment]
My concern is only about chaining functions, yes. With `iterable()` interface the issue is not there.

I specifically don't fancy these checks `is_array($iterable) ? $withoutKeys->asArray() : $withoutKeys;` as it adds some overhead for arrays. While the idea of returning the same type as was the input type is 👌🏾   we can't achieve it anyway because there are infinite possible types that are `iterable` (for example I'm heavy Ds user). So to me having different behaviour for arrays does not make much sense TBH.

Passing `Ds\Set` won't return me `Set` but `Generator`. Which is completely fine as I should rely on `iterable` and not `Set`. Therefore, to me there's no reason to be fixed on returning `array` for input `array` type. I even can't use the return value as `array` because it's covariant to `iterable` return type.

**Comment by JoshuaBehrens** [comment]
Knock knock. Fellow iterable-functions user here. Any help needed to get 2.0 merged? We really want to use the new features :)

**Comment by bpolaszek** [comment]
Hey @JoshuaBehrens - thanks for pinging!

@simPod I added a few commits based on your comments. Can you please review?

File: phpcs.xml.dist [hunk 1]
```diff
@@ -18,6 +18,7 @@
         <exclude name="Generic.Formatting.MultipleStatementAlignment"/>
         <exclude name="Squiz.Commenting.FunctionComment.SpacingAfterParamType"/>
         <exclude name="SlevomatCodingStandard.Functions.StaticClosure"/>
+        <exclude name="SlevomatCodingStandard.ControlStructures.RequireNullCoalesceEqualOperator.RequiredNullCoalesceEqualOperator"/>
```
Comment (simPod) [inline phpcs.xml.dist]: Do we want to support PHP 7.3 still? It EOLs in few months (6 Dec 2021)

**Review by simPod — APPROVED** [review]

File: phpcs.xml.dist (hunk 1 above)
Comment (bpolaszek) [reply phpcs.xml.dist]: Good point, I totally agree.

**Comment by bpolaszek** [comment]
Here it is: https://github.com/bpolaszek/php-iterable-functions/releases/tag/2.0

**Comment by JoshuaBehrens** [comment]
Gonna upgrade #soon

**Pull request merged**